### PR TITLE
Changes needed for torch 2.9.0 and transformers 4.57.1 updates

### PR DESCRIPTION
**Pull request by jazpurTT**
### Ticket
https://github.com/tenstorrent/tt-xla/issues/1020

### Problem description
tt-xla is updating the transformers library to 4.57.1 and torch to 2.9.0, we need to make changes on how we load some of the transformers models based on the updated in the library since our last update

### What's changed
1. Transformers changed how they process text/image inputs for Fuyu models, so we don't longer need to `model_inputs["image_patches"][0]`
2. Transformers changed the location of the `ALL_PARALLEL_STYLES` global variable
3. New transformers cache logic causing SEGFAULT issues if the model uses sliding window attention, and sliding window causing recompilation of ministral decode graph on each iteration. Changed to full attention.
4. Force padding to `right` by default.
5. _supports_sdpa is not set by default before `__init__` so we need to add property for OpenVLA loader
6. whisper requirements uplifted to torchcodec==0.8 to match compatibility with torch==2.9.0
7. gemma model drops pcc when padding inputs, removing it from input loader

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: gemma/pytorch/loader.py [hunk 1]
```diff
@@ -178,18 +179,11 @@ def load_inputs(
...
-                    inputs[key] = cast_input_to_type(inputs[key], dtype_override)
-            padded_input_ids, seq_len = pad_inputs(inputs["input_ids"], max_new_tokens)
-            padded_attention_mask, _ = pad_inputs(
-                inputs["attention_mask"], max_new_tokens
-            )
-            self.seq_len = seq_len
-            inputs["input_ids"] = padded_input_ids
-            inputs["attention_mask"] = padded_attention_mask
+                inputs[key] = cast_input_to_type(inputs[key], dtype_override)
         return inputs
```
[the remark this answers is not in the document]
Comment (jazpurTT) [reply gemma/pytorch/loader.py]: Good catch, thanks. Added an explanation on why this change is necessary in the PR desc
